fluxcal.py

```diff
@@ -7,25 +7,10 @@
 import argparse
 import os.path
 import numpy as np
-import logging
-import glob
 from shutil import copyfile, rmtree
-import matplotlib.pyplot as plt
-import pandas as pd
-import psrchive as ps
-from scipy import stats
 
 import getopt
 
-# WARNING - This next instruction is horrible code and I don't condone ever doing it again
-# Unfortunately, my hand is forced by a lack of other apparent options
-# Andrew Cameron - 10/02/2022
-
-sys.path.insert(1, sys.path[len(sys.path)-1])
-import astropy
-
-# end horrible code
-
```

Comment:
I'd like to address how you fixed this before merging everything all together. While this code may work on NT, I put this code in place to overcome a specific shortcoming on OzStar and I'm hesistant to take it out until that has been resolved.

Reply:
Like I said in the call, this worked on NT. Can you remember what the above code does as I'm not even sure

Reply:
I needed to use a function available only in a recent version of astropy, but the path to the package was at the end of my PYTHONPATH variable and was getting overridden by something higher up the chain, hence the import command was loading the wrong version. I couldn't figure out how to resolve the issue, so I implemented this command, which manually manipulated the path to make sure that the correct astropy version was being given priority.

As soon as I can run jobs again, I'll see whether the problem is still active.

Reply:
So, a few developments:

- I've dug up why it was I needed to make this change in the first place. Fluxcal.py needs to calculate a sky temperature for the location of the pulsar, so it has to extract the pulsar's location from a .par file. If the coordinates are in Ecliptic format (ELONG/ELAT), it converts them to RAJ/DECJ using Astropy, as part of the `get_radec_new()` function.
- Originally, this conversion accepted the ELONG/ELAT values in the "geocentricmeanecliptic" coordinate frame. However, this was only available in Astropy 4, while the most up-to-date version on OzStar was some instance of Astropy 3.
- Because I didn't have access to modify the module libraries, I had to install my own Astropy copy into my virtual environment and I needed to hack the PYTHON PATH to make sure it used the correct version.
- However, I have since changed the conversion routine to use "geocentrictrueecliptic" coordinates. This frame exists in older Astropy versions such that the hack is no longer needed.

From all of this, there are two key takeaways:

1. It looks as though this change will be fine. I'll want to double check it once oz005 becomes functional again, but it _should_ be okay.
2. It is unclear which coordinate frame is the "correct" frame. Using the **true** ecliptic frame gives an RA/DEC position about 1.8 arcseconds off what PSRCAT says it should be. For flux calibration this should be fine, but you may wish to chase up what the correct methodology is here because right now I'm not sure we've got this 100% right.

Reply:
I've run a few tests, and MeerPipe can successfully run with this code removed. But it remains to be seen whether the current ecliptic coordinate frame is the correct one to be using in this instance.